raiden_contracts/tests/test_channel_close.py

```diff
@@ -181,6 +181,56 @@ def test_close_wrong_sender(
         ).transact({'from': C})
 
 
+def test_close_nonce_zero(
+        get_accounts,
+        token_network,
+        create_channel,
+        create_balance_proof,
+):
+    (A, B) = get_accounts(2)
+    vals_B = ChannelValues(
+        deposit=20,
+        transferred=5,
+        locksroot=fake_bytes(32, '03'),
+        claimable_locked=3,
+        unclaimable_locked=4,
+        nonce=0,
+    )
+    # Create channel and deposit
+    channel_identifier = create_channel(A, B)[0]
+
+    # Create balance proofs
+    balance_proof_B = create_balance_proof(
+        channel_identifier,
+        B,
+        vals_B.transferred,
+        vals_B.locked,
+        vals_B.nonce,
+        vals_B.locksroot,
+    )
+
+    token_network.functions.closeChannel(
+        channel_identifier,
+        B,
+        *balance_proof_B,
+    ).transact({'from': A})
+
+    # Even though we somehow provide valid values for the balance proof, they are not taken into
+    # consideration if the nonce is 0.
+    # The Raiden client enforces that the nonce is > 0 if off-chain transfers are made.
+    (
+        _, _,
+        B_is_the_closer,
+        B_balance_hash,
+        B_nonce,
+        _,
+        _,
+    ) = token_network.functions.getChannelParticipantInfo(channel_identifier, B, A).call()
+    assert B_is_the_closer is False
+    assert B_balance_hash == EMPTY_BALANCE_HASH
+    assert B_nonce == 0
```

Comment:
You might want to duplicate these checks before the `close` call as well, but not required IMO.

Reply:
Added them.